Importing Libraries

In [2]:
import numpy as np
import pandas as pd

Taking dataset from scikit-Learn framework .

In [3]:
from sklearn.datasets import load_iris
iris_dataset = load_iris()

Exploring the dataset which we have imported from sklearn.datasets.

In [5]:
print("Keys in iris dataset : {}".format(iris_dataset.keys()))

Keys in iris dataset : dict_keys(['data', 'target', 'target_names', 'DESCR', 'feature_names', 'filename'])


In [6]:
print(iris_dataset['DESCR'][:200] + "\n ...")

.. _iris_dataset:

Iris plants dataset
--------------------

**Data Set Characteristics:**

    :Number of Instances: 150 (50 in each of three classes)
    :Number of Attributes: 4 numeric, predictive
 ...


In [7]:
print("Target names : {}".format(iris_dataset['target_names']))

Target names : ['setosa' 'versicolor' 'virginica']


In [8]:
print("Feature names : {}".format(iris_dataset['feature_names']))

Feature names : ['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)']


In [10]:
print("Type of data : {}".format(type(iris_dataset['data'])))
print("Shape of data : {}".format(iris_dataset['data'].shape))

Type of data : <class 'numpy.ndarray'>
Shape of data : (150, 4)


There are several keys present in iris data set.
"DESCR" key contains short description about the dataset.
"Target names" key contains names of differnt species of flowers.
"Feature names" key have different features of flowers like length and width of 
sepals and petals.
Shape of data tells us that the data contains of 150 rows and 4 columns.
Type of data is nd array
Printing out some samples of data.

In [11]:
print("Printing out some columns of data :\n {}".format(iris_dataset['data'][:5]))

Printing out some columns of data :
 [[5.1 3.5 1.4 0.2]
 [4.9 3.  1.4 0.2]
 [4.7 3.2 1.3 0.2]
 [4.6 3.1 1.5 0.2]
 [5.  3.6 1.4 0.2]]


In [12]:
print("Type of target : {}".format(type(iris_dataset['target'])))
print("Shape of target : {}".format(iris_dataset['target'].shape))

Type of target : <class 'numpy.ndarray'>
Shape of target : (150,)


Type of value stored in "target" key is nd array.
Shape of value stored in target key is (150,) which means it is one dimensional array.

In [13]:
print("Target : \n {}".format(iris_dataset['target']))

Target : 
 [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 2 2 2 2 2 2 2 2 2 2 2
 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2
 2 2]


0 is assigned for sentosa . 1 is assigned for versicolor . 2 is assigned for verginica .

TRAINING AND TESTING DATA :

In [14]:
from sklearn.model_selection import train_test_split
X_train , X_test , y_train , y_test = train_test_split(
    iris_dataset['data'] , iris_dataset['target'] , random_state = 0)

In [15]:
print("X_train shape : {}".format(X_train.shape))
print("y_train shape : {}".format(y_train.shape))
print("X_test shape : {}".format(X_test.shape))
print("y_test shape : {}".format(y_test.shape))

X_train shape : (112, 4)
y_train shape : (112,)
X_test shape : (38, 4)
y_test shape : (38,)


The output of train_test_split function is of type Numpy arrays .
Training data consist of 75% of dataset and testing data consist of 25% data of dataset . 

Making a Machine learning model using KNN .

In [44]:
from sklearn.neighbors import KNeighborsClassifier
knn = KNeighborsClassifier(n_neighbors = 1)

Training the model on training data.

In [45]:
knn.fit(X_train , y_train)

KNeighborsClassifier(algorithm='auto', leaf_size=30, metric='minkowski',
                     metric_params=None, n_jobs=None, n_neighbors=1, p=2,
                     weights='uniform')

Making Predictions

In [46]:
X_new = np.array([[5,2.9,1,0.2]])
print("X_new shape : {}".format(X_new.shape))

X_new shape : (1, 4)


Getting result of predictions : 

In [47]:
prediction = knn.predict(X_new)
print("Prediction : {}".format(prediction))
print("Predicted target name : {}".format(
    iris_dataset['target_names'][prediction]))

Prediction : [0]
Predicted target name : ['setosa']


Evaluating the model :

In [48]:
y_predict = knn.predict(X_test)
print("Test set prediction : \n {}".format(y_predict))

Test set prediction : 
 [2 1 0 2 0 2 0 1 1 1 2 1 1 1 1 0 1 1 0 0 2 1 0 0 2 0 0 1 1 0 2 1 0 2 2 1 0
 2]


Getting accuracy :

In [49]:
print("Test set score : {}".format(np.mean(y_predict == y_test)))

Test set score : 0.9736842105263158
